### PR TITLE
Migrate the route "/api/invocations/{invocation_id}" of the workflows API to Fast API

### PR DESCRIPTION
**Pull request by heisner-tillman**
This is a part of #10889.

## Summary
- [x] Refactored API route
- [ ] Annotated input parameters
- [ ] Operation now returns pydantic model
- [ ] Removed the mapping to the legacy route

## How to test the changes?
- [x] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
    You can find the interactive API documentation here: [http://127.0.0.1:8080/api/docs#/workflows/show_invocation_api_invocations__invocation_id__get](url)
image

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by martenson** [comment]
Hi @heisner-tillman, I did not get close look yet but please be aware of potential duplicate effort due to https://github.com/galaxyproject/galaxy/pull/16707.

File: lib/galaxy/schema/invocation.py [hunk 1]
```diff
@@ -153,6 +163,47 @@ class GenericInvocationEvaluationWarningWorkflowOutputNotFound(
...
+class InvocationState(str, Enum):
+    NEW = "new"  # Brand new workflow invocation... maybe this should be same as READY
+    READY = "ready"  # Workflow ready for another iteration of scheduling.
+    SCHEDULED = "scheduled"  # Workflow has been scheduled.
+    CANCELLED = "cancelled"
+    CANCELLING = "cancelling"  # invocation scheduler will cancel job in next iteration
+    FAILED = "failed"
+
+
+class EncodedInvocation(BaseModel):
```
Comment (mvdbeek) [inline lib/galaxy/schema/invocation.py:177]: Can you name these things `{Something}Response` ?

File: lib/galaxy/schema/invocation.py [hunk 2]
```diff
@@ -153,6 +163,47 @@ class GenericInvocationEvaluationWarningWorkflowOutputNotFound(
...
+    # TODO - Add proper models
+    steps: Optional[List[Dict[str, Optional[str]]]] = None
+    inputs: Optional[Dict[str, Dict[str, Optional[str]]]] = None
+    input_step_parameters: Optional[Dict[str, Dict[str, Optional[str]]]] = None
+    outputs: Optional[Dict[str, Dict[str, Optional[str]]]] = None
+    output_collections: Optional[Dict[str, Dict[str, Optional[str]]]] = None
+    output_values: Optional[Dict[str, Optional[str]]] = None
+    # TODO understand where this comes from
+    message: Optional[str] = Field(
+        default=None, title="Message", description="Message associated with this invocation."
```
Comment (mvdbeek) [inline lib/galaxy/schema/invocation.py:203]: Are these really optional, or is it just that they can be null ? That's not the same

**Comment by heisner-tillman** [comment]
To avoid duplication, this work will be continued in collaboration with @martenson on his pull request #16707!

**Pull request closed without merging**